sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/CustomModelTransforms.java

```diff
@@ -0,0 +1,117 @@
+                .map(trainingDocumentItem -> new TrainingDocumentInfo(
+                    trainingDocumentItem.getDocumentName(),
+                    TrainingStatus.fromString(trainingDocumentItem.getStatus().toString()),
+                    trainingDocumentItem.getPages(),
+                    new IterableStream<FormRecognizerError>(transformTrainingErrors(trainingDocumentItem.getErrors()))))
+                .collect(Collectors.toList());
+
+        List<CustomFormSubModel> subModelList = new ArrayList<>();
+        String formType = "form-";
+        // unlabeled model
+        if (modelResponse.getKeys() != null) {
+            Map<String, CustomFormModelField> fieldMap = new TreeMap<>();
+            modelResponse.getKeys().getClusters().forEach((clusterKey, clusterFields) -> {
+                for (int i = 0, clusterFieldsSize = clusterFields.size(); i < clusterFieldsSize; i++) {
+                    String eachField = clusterFields.get(i);
+                    String fieldLabel = "field-" + i;
+                    fieldMap.put(fieldLabel, new CustomFormModelField(fieldLabel, eachField, null));
+                }
+                subModelList.add(new CustomFormSubModel(
+                    null,
+                    fieldMap,
+                    formType + clusterKey));
+            });
+        } else if (modelResponse.getTrainResult().getFields() != null) {
+            // labeled model
+            Map<String, CustomFormModelField> fieldMap = new TreeMap<>();
+            modelResponse.getTrainResult().getFields()
+                .forEach(formFieldsReport -> fieldMap.put(formFieldsReport.getFieldName(),
+                    new CustomFormModelField(null, formFieldsReport.getFieldName(),
+                        formFieldsReport.getAccuracy())));
+            subModelList.add(new CustomFormSubModel(
+                modelResponse.getTrainResult().getAverageModelAccuracy(),
+                fieldMap,
+                formType + modelInfo.getModelId()));
+        }
+
+        return new CustomFormModel(
+            modelInfo.getModelId().toString(),
+            ModelTrainingStatus.fromString(modelInfo.getStatus().toString()),
+            modelInfo.getCreatedDateTime(),
+            modelInfo.getLastUpdatedDateTime(),
+            new IterableStream<>(subModelList),
+            new IterableStream<FormRecognizerError>(
+                transformTrainingErrors(modelResponse.getTrainResult().getErrors())),
+            new IterableStream<TrainingDocumentInfo>(trainingDocumentInfoList));
+    }
+
+    /**
+     * Helper method to convert the list of {@link ErrorInformation} to list of {@link FormRecognizerError}.
+     *
+     * @param trainingErrorList The list of {@link ErrorInformation}.
+     *
+     * @return The list of {@link FormRecognizerError}
+     */
+    private static List<FormRecognizerError> transformTrainingErrors(List<ErrorInformation> trainingErrorList) {
```

Comment:
Lets have this return `IterableStream` instead, I don't see the `List` return being used for anything other than constructing an `IterableStream`. With this change we could optimize the `null` or empty case to return `IterableStream.of(null)` which optimizes to a static immutable instance.